diags/diag_nvme.c

```diff
@@ -375,9 +375,35 @@ static int diagnose_nvme(char *device_name, struct notify *notify, char *file_pa
 	char endurance_s[sizeof(vpd.endurance) + 1], capacity_s[sizeof(vpd.capacity)+1];
         uint64_t event_id;
 	uint8_t severity;
+	FILE *fp;
+	char tr_file_path[PATH_MAX];
 	uint32_t raw_data_len = 0;
 	unsigned char *raw_data = NULL;
 
+	/*
+	 * Skip diag test if NVMe is connected over fabric
+	 */
+	snprintf(tr_file_path, sizeof(tr_file_path),
+			NVME_SYS_PATH"/%s/%s", device_name, "transport");
+	fp = fopen(tr_file_path, "r");
+	if (fp) {
+		char buf[32];
```

Comment:
It looks like you are only interested in first 4 char 'pcie' from the file. Wouldn't buf[5] is enough ?